### ISSUE TITLE
[question] Why tf.gradient_stops in SAC code?

### ISSUE THREAD

**Issue by Antymon** [hill-a/stable-baselines#564]
There are 3 places where `tf.gradient_stop` is used: q_backup, v_backup and ent_coeff. I can't see why we would need to block out v_target variables contribution to the q network's gradients, or whatever else's contribution when calculating ent_coeff gradient wrt single variable log_ent_coeff. Can someone relate, please?

**Comment by araffin** [comment]
Hello,

> I can't see why we would need to block out v_target variables contribution to the q network's gradients

Good remark. The short answer is that it is for clarity to explicitly show what is updated by each loss.


In practice, this is not mandatory here because of the `var_list` argument:
https://github.com/hill-a/stable-baselines/blob/a1ab7a1c2903e7e1c38756d8cdf7a54a5fd5781e/stable_baselines/sac/sac.py#L269
which means that the gradients are computed only for those variables.


Otherwise, you would for instance update the entropy coeff using the policy loss, which is not how you should update it. The same goes for the `v_backup`, this is meant to be a fixed target, not something that you can update, hence the `tf.stop_gradient`.
